pkg/core/plugin.go

```diff
@@ -0,0 +1,30 @@
+package core
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+)
+
+var plugins = make(map[string]Plugin)
+
+func Register(name string, plugin Plugin) {
```

Comment:
So basically instead of using go mod to inexplicitly 



This is clever, but I want to know what we getting from this? Is this art for the art or there is any benefit coming from
registration of plugins - which is extra hassle for something that can be done using golang and cobra (importing root command and adding it to the list in any place we want.

Reply:
Now that I think, yes, registering plugins isn't adding any extra value to our SDK

Reply:
As mentioned in discussions we can always go back to this spikes. Sometimes we need to let go and move on. Lets close this PR, create issues we spoke and push forward with whatever will be the most challenging for you.

Reply:
Not having any ideas around 3am is because it is 3am and not because this approach is bad. Get some sleep and come back to this later

Reply:
Yes, you're absolutely right.

Will close this PR for now and shift my focus to the current issues/discussions for the time being